### PR TITLE
sv-lang: 3.0 -> 6.0

### PR DESCRIPTION
**Pull request by thoughtpolice**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by dotlambda** [comment]
See https://github.com/NixOS/nixpkgs/pull/292709 for the catch2_3 update. We need to make sure it builds on Darwin.

**Comment by SharzyL** [comment]
Why so many rebuilds?

**Comment by thoughtpolice** [comment]
> See #292709 for the catch2_3 update. We need to make sure it builds on Darwin.

Good to know. I updated catch2_3 because the sv-lang 6.0 CMakeLists.txt file specifies it, but apparently 3.5.2 works still works just fine. Therefore, I've dropped this patch from the PR.

> Why so many rebuilds?

🤷🏻 More things transitively use `catch2_3` than you might expect, I guess. I didn't think it would be so big, either.

File: pkgs/applications/science/electronics/sv-lang/default.nix [hunk 1]
```diff
@@ -55,14 +39,11 @@ stdenv.mkDerivation rec {
...
-    unordered_dense
     boost182
-    fmt_9
+    fmt_10
   ];
 
-  # TODO: a mysterious linker error occurs when building the unittests on darwin.
-  # The error occurs when using catch2_3 in nixpkgs, not when fetching catch2_3 using CMake
-  doCheck = !stdenv.isDarwin;
+  doCheck = true;
```
Comment (dotlambda) [inline pkgs/applications/science/electronics/sv-lang/default.nix:46]: Did you try Darwin?

Comment (thoughtpolice) [reply pkgs/applications/science/electronics/sv-lang/default.nix:46]: Actually, it was pre-emptive: I was optimistically hoping to let the CI system run it on my behalf so I could avoiding rebooting over from Asahi Linux into macOS. :) If it fails I'll take a peek and look into it or revert this, rest assured.

**Comment by wegank** [comment]
Superseded by #355401.

**Pull request closed without merging**